pyproject.toml

```
@@ -1,47 +1,206 @@
+dev = [
+    "black",
+    "hyperspy[all]",
+    "hyperspy[coverage]",
+    "hyperspy[doc]",
+    "hyperspy[tests]",
+]
+doc = [
+    "numpydoc",
+    "pydata_sphinx_theme",
+    "setuptools_scm",
+    "sphinx-copybutton",
+    "sphinx-design",
+    "sphinx-favicon",
+    "sphinx-gallery",
+    "sphinx>=1.7",
+    "sphinxcontrib-mermaid",
+    "sphinxcontrib-towncrier>=0.3.0a0",
+    "towncrier",
+]
+gui-jupyter = [
+    "hyperspy_gui_ipywidgets>=2.0",
+    "ipympl",
+]
+gui-traitsui = [
+    "hyperspy_gui_traitsui>=2.0",
+]
+ipython = [
+    "IPython>7.0,!=8.0",
+    "ipyparallel",
+]
+learning = [
+    "scikit-learn>=1.0.1",
+]
+speed = [
+    "numba",
+    "numexpr",
+]
+tests = [
+    "pytest-instafail",
+    "pytest-mpl",
+    "pytest-rerunfailures",
+    "pytest-xdist",
+    "pytest>=3.6",
+    "setuptools_scm"
+]
+
+[project.urls]
+"Bug Reports" = "https://github.com/hyperspy/hyperspy/issues"
+Documentation = "https://hyperspy.org/hyperspy-doc/current/index.html"
+Conda-forge = "https://anaconda.org/conda-forge/hyperspy"
+Homepage = "https://hyperspy.org"
+Support = "https://gitter.im/hyperspy/hyperspy"
+Source = "https://github.com/hyperspy/hyperspy"
+
+[tool.coverage.report]
+precision = 2
+
+[tool.coverage.run]
+branch = true
+omit = [
+    "examples/*",
+    "hyperspy/conftest.py",
+    "hyperspy/tests/*",
+    "setup.py",
+]
+source = ["hyperspy"]
+
+[tool.pytest.ini_options]
+addopts = "-ra -n auto --dist loadfile"
+filterwarnings = [
+    "ignore:Matplotlib is currently using agg:UserWarning",
+]
+testpaths = ["hyperspy/tests"]
+xfail_strict = false
+
+[tool.setuptools.package-data]
+"*" = [
+    "hyperspy_extension.yaml",
+    "misc/dask_widgets/*.html.j2",
+]
+
+[tool.setuptools.packages.find]
+include = ["hyperspy*"]
+where = ["."]
+
+[tool.setuptools.exclude-package-data]
+"*" = ["*.png"]
+
+[tool.setuptools_scm]
+# Presence enables setuptools_scm, the version will be determine at build time from git
 
 [tool.towncrier]
-package = "hyperspy"
-filename = "CHANGES.rst"
 directory = "upcoming_changes/"
-# see https://github.com/twisted/towncrier/issues/346
-#title_format = "v{version} ({project_date})"
+filename = "CHANGES.rst"
 issue_format = "`#{issue} <https://github.com/hyperspy/hyperspy/issues/{issue}>`_"
-
-  [[tool.towncrier.type]]
-  directory = "new"
-  name = "New features"
-  showcontent = true
-
-  [[tool.towncrier.type]]
-  directory = "bugfix"
-  name = "Bug Fixes"
-  showcontent = true
-
-  [[tool.towncrier.type]]
-  directory = "doc"
-  name = "Improved Documentation"
-  showcontent = true
-
-  [[tool.towncrier.type]]
-  directory = "deprecation"
-  name = "Deprecations"
-  showcontent = true
-
-  [[tool.towncrier.type]]
-  directory = "enhancements"
-  name = "Enhancements"
-  showcontent = true
-
-  [[tool.towncrier.type]]
-  directory = "api"
-  name = "API changes"
-  showcontent = true
-
-  [[tool.towncrier.type]]
-  directory = "maintenance"
-  name = "Maintenance"
-  showcontent = true
+package = "hyperspy"
```

In the other packages, this line is `package_dir`

Also the line `title_format = "{version} ({project_date})" package_dir = "hyperspy"` is missing?

---

```suggestion
package = "hyperspy"
title_format = "{version} ({project_date})"
```

For consistency, should we remove all the `v` (for example in `v2.0rc0`) from the release notes?

> In the other packages, this line is `package_dir`

According to https://towncrier.readthedocs.io/en/stable/configuration.html#extra-top-level-keys-for-python-projects, we should use `package`.

---

Well it seems to be two different things, but the extensions currently define only `package_dir`, while here only `package` is defined.

---

> For consistency, should we remove all the `v` (for example in `v2.0rc0`) from the release notes?

I always like consistency :-)